### DISCUSSION TITLE
How to execute `shouldBehaveLikeGreeter` with a parameter

### DISCUSSION THREAD

**Discussion by SvenMeyer** [PaulRBerg/hardhat-template#105] [Q&A]
So far I am very pretty happy with this template, have used it for quite some projects !

However, likely due to my lack in understanding of waffle and/or TypeScript, I can't get a simple extension to work (with my approach) where I just want to set a "global" variable during deployment or a previous test, and then pass it on as a parameter to `shouldBehaveLikeGreeter` - it always ends up as `undefined` there ... 

```typescript
import type { SignerWithAddress } from "@nomiclabs/hardhat-ethers/dist/src/signer-with-address";
import { artifacts, ethers, waffle } from "hardhat";
import type { Artifact } from "hardhat/types";

import type { Greeter } from "../../src/types/Greeter";
import { Signers } from "../types";
import { shouldBehaveLikeGreeter } from "./Greeter.behavior";

var testString: string; // <<<<<<<<<<<<<<<<<<<<<<

describe("Unit tests", function () {

  before(async function () {
    this.signers = {} as Signers;

    const signers: SignerWithAddress[] = await ethers.getSigners();
    this.signers.admin = signers[0];
  });

  describe("Greeter", function () {
    beforeEach(async function () {
      const greeting: string = "Hello, world!";
      const greeterArtifact: Artifact = await artifacts.readArtifact("Greeter");
      this.greeter = <Greeter>await waffle.deployContract(this.signers.admin, greeterArtifact, [greeting]);

      testString = "Hello, world!"; // <<<<<<<< does not work > undefined in shouldBehaveLikeGreeter
    });

    shouldBehaveLikeGreeter(testString); // <<<<<<<<<<<<<<<<<<<
  });

});

```

```typescript
import { expect } from "chai";

export function shouldBehaveLikeGreeter(parameter: string): void {
  it("should return the new greeting once it's changed", async function () {
    expect(await this.greeter.connect(this.signers.admin).greet()).to.equal(parameter);

    await this.greeter.setGreeting("Bonjour, le monde!");
    expect(await this.greeter.connect(this.signers.admin).greet()).to.equal("Bonjour, le monde!");
  });
}
```

https://github.com/SvenMeyer/test-parameter

**Comment by PaulRBerg** [comment]
Hi Sven, I'm glad you like this template and thanks so much for the many issues, PRs and suggestions you have made over time.

Unfortunately, there's no way to pass arguments the way you want to. This is not a limitation of this template, Hardhat or TypeScript. It's a limitation of Mocha. In short, you cannot pass the values generated in the low-level `beforeEach` or `it` contexts to the high-level contexts (`describe`). Mocha flushes the context after each test run, and the values generated in the lower-level contexts are reset. By the time the next higher-level context is run, the values are emptied out.
